### ISSUE TITLE
Improve documentationfor tower-test

### ISSUE THREAD

**Issue by alfiejfs** [tower-rs/tower#772]
There is currently not a lot of documentation around the tower-test library, with minimal descriptions and no usage examples. Hopefully this shouldn't be too hard to improve. I found it very difficult to use the library as I could not find too many examples online and am not yet familiar with the tower ecosystem

**Comment by an-data** [comment]
I agree with you, It's difficult to impl some fn  by tower crate, If there are many examples, it  will be easy to use

**Comment by an-data** [comment]
> There is currently not a lot of documentation around the tower-test library, with minimal descriptions and no usage examples. Hopefully this shouldn't be too hard to improve. I found it very difficult to use the library as I could not find too many examples online and am not yet familiar with the tower ecosystem

I agree with you, It's difficult to impl some fn by tower crate, If there are many examples, it will be easy to use
